Python/abiinfo.c

```diff
@@ -0,0 +1,100 @@
+/* ABI information object implementation */
+
+#include "Python.h"
+#include "pycore_initconfig.h"    // _PyStatus_OK()
+#include "pycore_structseq.h"     // _PyStructSequence_InitBuiltin()
+
+static PyTypeObject AbiInfoType;
+
+PyDoc_STRVAR(abi_info__doc__,
+"ABI information object.\n\
+This object provides information about the ABI \
+features available in the current Python interpreter.\n");
+
+static PyStructSequence_Field abi_info_fields[] = {
```

Comment:
`PyStructSequence` is usually used when adding named fields to something that used to be a plain tuple.
The ABI info is new; it does not need to be tuple-line (iterable, indexable with integers). It can be a simple namespace.


See `sys.implementation` (`make_impl_info` in `sysmodule.c`) for an example.